### PR TITLE
Custom setproctitle

### LINKED ISSUE #32660: [$100 bounty] Remove `setproctitle` dependency
Replace `setproctitle` with our own small helper in `common/`. Just needs to work on Linux.

### PR DESCRIPTION
**Pull request by schlimeszn**
Resolves: #32660  

It's a simple refactor with a test to confirm correctness.

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Doesn't seem to completely work. Name change isn't reflected in `htop` anymore. Just run `selfdrive/modeld/modeld` to check.

**Comment by schlimeszn** [comment]
> Doesn't seem to completely work. Name change isn't reflected in `htop` anymore. Just run `selfdrive/modeld/modeld` to check.

In ```$HOME/.config/htop/htoprc``` set ```show_thread_names=1```

Is this a valid fix? Spent an hour trying to modify the process names in memory with ctypes and no success. Looked at setproctitle source code and would rather flip a bool than steal that

**Comment by schlimeszn** [comment]
You can also ```F2``` in ```htop``` go to ```Display Options``` and select ```Show custom thread names```. If you want this change to persist, https://unix.stackexchange.com/a/181370. Or change the config file like I mentioned

**Comment by jyoung8607** [comment]
The `prctl` method sets thread names, not the process name. I glanced at the Python `setproctitle` source, and the method necessary to change the actual process title on Linux seems pretty ugly. If you truly want to change the process title, I don't think it makes much sense for openpilot to roll its own method, just keep using the module.

I think @schlimeszn has proposed a reasonable solution, it's just that AGNOS ships an `.htoprc` that isn't helping us here. It's configured to show individual threads within processes:

```
hide_threads=0
hide_kernel_threads=0
hide_userland_threads=0
```

But with the process name and not the thread name:

```
show_thread_names=0
```

I think it may be good to change that regardless. Additionally, it would be nice to be able to name threads within a multithreaded process and then be able to see which threads are consuming CPU in `htop`.

If we proceed with a thread naming approach, I have some notes:
* An agnos-builder PR should be filed to update the[ shipping `.htoprc`](https://github.com/commaai/agnos-builder/blob/master/userspace/home/.config/htop/htoprc)
* Rename the functions to reflect setting the thread name as opposed to process name
* When setting, PR_SET_NAME has length and null termination constraints that should be checked
* When reading, the name would need to come from `/proc/PID/task/TID/comm` instead of `/proc/PID/comm`

**Comment by adeebshihadeh** [comment]
Let's go with what @jyoung8607's outlined. I think this is a reasonable compromise to get rid of a dependency for something that provides minimal value.

File: common/threadname.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+LINUX = os.name == 'posix' and os.uname().sysname == 'Linux'
+
+if LINUX:
+  libc = ctypes.CDLL('libc.so.6')
+
+def setthreadname(name: str) -> None:
+  if LINUX:
+    if len(name)>15:
+      name = name[:15]
+    name += '\0'
```
Comment (adeebshihadeh) [inline common/threadname.py:13]:
```suggestion
    if len(name)>15:
      name = name[:15]
    name = name[:15] + '\0'
```

**Comment by adeebshihadeh** [comment]
Reverted on master since `test_onroad` needs some updates after this. I'll look into this more soon

**Pull request merged**